js/src/react/mapReactData.js

```diff
@@ -85,9 +85,10 @@ dataMappers.element = ({ module, name, props: propsData }) => {
 // Used to implement `setInput()` and `triggerEvent()` R functions. In case of `triggerEvent()`,
 // we have `argIdx === null` and the returned function just sets the Shiny input to `TRUE`
 // on every call (this works thanks to `priority: 'event'`).
-dataMappers.input = ({ id, argIdx }) => (
+dataMappers.input = ({ id, argIdx, accessor }) => (
   (...args) => {
-    const value = argIdx === null ? true : args[argIdx];
+    let value = argIdx === null || argIdx === undefined ? true : args[argIdx];
+    value = accessor === null || accessor === undefined ? value : eval(`${args}accessor`); // eslint-disable-line no-eval
```

Comment:
I find this hard to read. Let's have a separate data mapper for `triggerEvent()` to avoid `=== null` checks. Then consider a single `accessor` argument and check its type, or perhaps have separate data mappers for `argIdx` and `accessor` cases.

Reply:
I preferred using a more explicit if conditional, but that was giving me a weird issue, where `args` became `undefined`.

I'll improve on this

Reply:
I've unified the solution, so that `jsAccessor = 1` resolves to "jsAccessor[1]" in R. This keeps the code cleaner with a single point of maintenance.

`triggerEvent` has its own dataMapper that is straightforward